src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyTracker.cs

```diff
@@ -32,17 +33,21 @@ public class VmrDependencyTracker : IVmrDependencyTracker
     private const string DefaultVersion = "8.0.100";
 
     private readonly Lazy<AllVersionsPropsFile> _repoVersions;
+    private readonly SourceManifest _sourceManifest;
     private readonly string _allVersionsFilePath;
     private readonly IVmrInfo _vmrInfo;
+    private readonly IFileSystem _fileSystem;
 
     public IReadOnlyCollection<SourceMapping> Mappings { get; }
 
-    public VmrDependencyTracker(IVmrInfo vmrInfo, IReadOnlyCollection<SourceMapping> mappings)
+    public VmrDependencyTracker(IVmrInfo vmrInfo, IReadOnlyCollection<SourceMapping> mappings, IFileSystem fileSystem)
     {
         _vmrInfo = vmrInfo;
+        _fileSystem = fileSystem;
         _allVersionsFilePath = Path.Combine(vmrInfo.VmrPath, VmrInfo.GitInfoSourcesDir, AllVersionsPropsFile.FileName);
         _repoVersions = new Lazy<AllVersionsPropsFile>(LoadAllVersionsFile, LazyThreadSafetyMode.ExecutionAndPublication);
-
+        _sourceManifest = LoadSourceManifest();
```

Comment:
I think it's better to receive the `SourceManifest` directly just like we do with the mappings rather than do work in the constructor (usually, constructors should be very lightweight and not perform complex actions that can potentially crash).

It will then also be easier to test this class.

Reply:
Technically, you can also make another overload on the `SourceManifest` to accept path and create a stream there and call the existing one (do not use `FileSystem` for that but directly use `System.IO` - we will just not test that overload but it's fine as it will be small). You can then move the code loading the manifest to `VmrRegistrations` (where we load mappings already).